### ISSUE TITLE
externs for getters and setters

### ISSUE THREAD

**Issue by kennylerma** [apache/royale-typedefs#1]
Currently, the below javascript will produce a public variable in the generated ActionScript using externc.  How do I notate the javascript to become a getter or setter?

/**
 * @type {number}
 */
createjs.MovieClip.prototype.currentFrame;

**Comment by aharui** [comment]
It looks like the compiler only generates getter/setters if the variable is also implemented in an interface.  That's because interfaces can only have getter/setter functions and not vars.

In theory, code calling into the external class shouldn't care whether it is a var or getter/setter.  What is the use case where it matters?

**Comment by kennylerma** [comment]
In my particular use case, I need to override a getter method.  Additionally, I want to match the external library as closely as possible to catch compile time errors like attempting to set a read only variable.

I hoping I can get contribute a bit to this repo by completing out the CreateJS library, but I'll need ability to create the getters and setters.

**Comment by kennylerma** [comment]
Ah, I guess I should have expected it to be more complicated. :)

This is how CreateJS sets up its getters/setters
```
Object.defineProperties(p, {
	totalFrames: { get: p._getDuration }
});
```

So, Royale would need to generate new methods to assign the getters/setters to and keep the Superclass methods in-tact.
```
p._customTotalFrames = function() {
	console.log("Superclass value: " + this._getDuration());
	return 200; // return custom Subclass value.
}

Object.defineProperties(p, {
	totalFrames: { get: p._customTotalFrames }
});
```

**Comment by kennylerma** [comment]
Yeah, CreateJS has it's own utility class for overriding methods which of course wouldn't be used here.

However, I was thinking you could using something like this to call the super class method.
```
Object.getOwnPropertyDescriptor(obj, 'MyField').get;  // returns the superclass get function

/* setter */
Object.getOwnPropertyDescriptor(obj, 'MyField').set;  // returns the superclass set function

// so super.MyField would return the original method.
super.MyField -> Object.getOwnPropertyDescriptor(obj, 'MyField').get;
```
Could that work?

**Comment by aharui** [comment]
The compiler can be told to output different patterns.  So if there is a pattern to how you use a CreateJS utility class, a custom "emitter" for CreateJS could do so.

I could use the pattern you describe above, but I think we used to output something like that and found it was very slow.  But maybe that's as good as it gets.  IIRC, the actual pattern has to be a bit more complex in the case where the superclass doesn't have the API that is being overridden, the superclass's superclass does.  I think the pattern has to see if the Object.getOwnPropertyDescriptor return null/undefined and then loop through the ancestor classes.

A second problem is knowing what name to give to your override.  Currently we generate a get__foo and set__foo (yes, two underscores) but we don't have any guarantee that someone else might have already used that name in one of the base classes.

If you can manually change some of the APIs to use a proposed pattern of Object.getOwnPropertyDescriptor so you know it works, then we know if we teach the compiler to output the same that it will be worth the effort.

And bonus if you or some other volunteer wants to learn enough about how the compiler code generator works to make these changes.

**Comment by kennylerma** [comment]
Thanks @aharui, I'm going to create a working example and see what works best.  I've been trying to learn about compilers lately and I'm certainly interested in the Royale Compiler.  I'll have to try building it myself and see how far I get.
